straxen/plugins/event_patternfit.py

```diff
@@ -305,7 +331,27 @@ def compute_s2_llhvalue(self, events, result):
                     store_2LLH_ch = np.zeros((norm_llh_val.shape[0], self.config['n_top_pmts']) )
                     store_2LLH_ch[:, self.pmtbool_top] = norm_llh_val
                     result[t_+'_2llh_per_channel'][cur_s2_bool] = store_2LLH_ch
-                    
+
+    def compute_s2_neural_llhvalue(self, events, result):
+        for t_ in ['s2', 'alt_s2']:
+            # Selecting S2s for pattern fit calculation
+            # - must exist (index != -1)
+            # - must have total area larger minimal one
+            # - must have positive AFT
+            x, y = events[t_ + '_x'], events[t_ + '_y']
+            cur_s2_bool = (events[t_ + '_area'] > self.config['s2_min_area_pattern_fit'])
+            cur_s2_bool &= (events[t_ + '_index'] != -1)
```

Comment:
It's not from this PR, but isn't that redundant with the minimum S2 area requirement?

Reply:
I didn't quite get the line of 343 actually. In which case can an s2_index be -1?

Reply:
If s2 doesn't exist, then `s2_index` is -1. However, since we already impose a minimum area for s2, the existence requirement seems to be redundant to me.

Reply:
Sure, thanks for pointing this out! I deleted this line and tested it.